### PR TITLE
Handle long splices for textfield and listfield.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Using `Array.splice` when the to-be-spliced values can run into browser limits for the most arguments allowed to be passed to a variadic function. This can happen especially when initializing a large text field (i.e., a few hundred thousand characters).

This PR implements an internal utility for splitting up large splices, getting around the browser limitations on the number of function args. The chunk size 100000 comes from some experimentation with Chrome and Firefox.

Now, text files of this size also start running into problems with ID sizes, which we still need to solve. But this is a mostly orthogonal problem.

### CODE REVIEW

File: packages/datastore/src/textfield.ts [hunk 1]
```diff
@@ -641,4 +641,40 @@ namespace Private {
...
+    }
+    let size = 100000;
+    if (items.length < size) {
+      return arr.splice(start, deleteCount || 0, ...items);
+    }
+    let deleted = arr.splice(start, deleteCount);
+    let n = Math.floor(items.length / size);
+    let idx = 0;
+    for (let i = 0; i < n; i++, idx += size) {
+      arr.splice(idx, 0, ...items.slice(idx, size));
```
Comment (vidartf) [inline packages/datastore/src/textfield.ts]: I might be reading this wrong, but shouldn't it be like this?
```suggestion
      arr.splice(start, 0, ...items.slice(idx, size));
```
A brief sanity check which checks that it behaves the same as `splice` might be reassuring.

Comment (ian-r-rose) [reply packages/datastore/src/textfield.ts]: Good catch! It shouldn't be `start` or `idx`, but instead something like `start + idx`.

Comment (ian-r-rose) [reply packages/datastore/src/textfield.ts]: I added a test, but not a very good one :/

File: packages/datastore/tests/tsconfig.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "moduleResolution": "node",
     "target": "ES5",
     "outDir": "./build",
-    "lib": ["ES5", "DOM"],
+    "lib": ["ES6", "DOM"],
```
Comment (sccolbert) [inline packages/datastore/tests/tsconfig.json]: Why this change? We are still compiling down to ES5.

Comment (ian-r-rose) [reply packages/datastore/tests/tsconfig.json]: This should still compile to ES5, due to the `target` field above, no? I don't recall why I wanted to bump this here (possibly a transient usage of an es6 feature).

**Comment by ian-r-rose** [comment]
Thanks for the review @blink1073. I'll need to make a follow-up here to resolve the logic error that @vidartf pointed out above.
